File: net.tenteCsharp/src-gen/lightManagement/RoomGUI.cs
using System;
using System.Collections;
using System.Text;

namespace SmartHome
{
	public partial class RoomGUI : TypeComponent
	{
		public LightNotifyPort lightNotifyPort;
			public  ArrayList listLightGUI= new ArrayList();
	
		public RoomGUI (String id) 
			: base(id)
		{
		}
		public LightNotifyPort getLightNotifyPort()
		{
			return lightNotifyPort;
		}
	
		public void setLightNotifyPort(RoomGUI.LightNotifyPort value)
		{
			this.lightNotifyPort=value;
		}
	
		public  ArrayList getListLightGUI()
		{
			return listLightGUI;
		}
	
		public void setListLightGUI(ArrayList value)
		{
			this.listLightGUI=value;
		}
		
		public void addListLightGUIElement(LightGUI value)
		{
			this.listLightGUI.Add(value);
		}
	

	
		
		public class LightNotifyPort : TypePort
		{
 		public ArrayList portsIGeneralLightNotify = new ArrayList();

			public LightNotifyPort()
				: base()
			{
				portsIGeneralLightNotify=new ArrayList();

			}


			public ArrayList getPortsIGeneralLightNotify()
			{
	        	return portsIGeneralLightNotify;
	        }
				
			public void connectPort(IGeneralLightNotify port)
			{
				portsIGeneralLightNotify.Add(port);
			}
	
		}
	}
}

File: net.tenteCsharp/src_code/lightManagement/RoomGUI.java


cclass lightManagement.LightManagement; 

public cclass RoomGUI extends TypeComponent{ 

	public LightNotifyPort lightNotifyPort;
		public ArrayList listLightGUI;
	
	
	public RoomGUI (String id){
		super(id);
	}
	
	public LightNotifyPort getLightNotifyPort(){
		return lightNotifyPort;
	}
	
	public void setLightNotifyPort(RoomGUI.LightNotifyPort value){
		this.lightNotifyPort=value;
	}
	
	public ArrayList getListLightGUI(){
		return listLightGUI;
	}
	
	public void setListLightGUI(ArrayList value){
		this.listLightGUI=value;
	}
	
	public void addListLightGUIElement(LightGUI value){
		this.listLightGUI.add(value);
	}
	

	
	
	public cclass LightNotifyPort extends TypePort{

		public ArrayList portsIGeneralLightNotify;

		public LightNotifyPort (){
			super();
			portsIGeneralLightNotify=new ArrayList();

		}


			
		public void connectPort(IGeneralLightNotify port){
			portsIGeneralLightNotify.add(port);
		}
	}

}